File: decompiled/3.17.1/be/ehealth/businessconnector/therlink/exception/TherLinkBusinessConnectorException.java
package be.ehealth.businessconnector.therlink.exception;

import be.ehealth.technicalconnector.exception.ConnectorException;
import java.text.MessageFormat;

public class TherLinkBusinessConnectorException extends ConnectorException {
   private static final long serialVersionUID = 1L;

   public TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues errorCodeValue, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode());
   }

   public TherLinkBusinessConnectorException(TherLinkBusinessConnectorExceptionValues errorCodeValue, Throwable e, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode(), e);
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/exception/TherLinkBusinessConnectorException.kt
package org.taktik.connector.business.therlink.exception

import org.taktik.connector.technical.exception.ConnectorException
import java.text.MessageFormat

class TherLinkBusinessConnectorException : ConnectorException {

	constructor(errorCodeValue: TherLinkBusinessConnectorExceptionValues, vararg params: Any) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode) {}

	constructor(errorCodeValue: TherLinkBusinessConnectorExceptionValues, e: Throwable, vararg params: Any) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode, e) {}

	companion object {
		private val serialVersionUID = 1L
	}
}
